bench/benchmarks/fps.js

```diff
@@ -0,0 +1,100 @@
+var measureFramerate = require('../lib/measure_framerate');
+
+var smallFeatureCollection = {
+    "type": "FeatureCollection",
+    "features": [{
+        "type": "Feature",
+        "properties": {},
+        "geometry": {
+            "type": "Point",
+            "coordinates": [ -77.032194, 38.912753 ]
+        }
+    }]
+};
+
+var largeFeatureCollection = require('../data/naturalearth-land.json');
+
+var DURATION_MILLISECONDS = 5 * 1000;
+
+module.exports = function(options) {
+    var evented = util.extend({}, Evented);
+
+    var map = options.createMap({
+        width: 1024,
+        height: 768,
+        zoom: 5,
+        center: [-77.032194, 38.912753],
+        style: 'mapbox://styles/mapbox/bright-v8'
+    });
+
+    map.on('load', function() {
+        map.addSource('geojson', {
+            'type': 'geojson',
+            'data': {
+                'type': 'FeatureCollection',
+                'features': []
+            }
+        });
+
+        map.addLayer({
+            'id': 'geojson-polygon',
+            'source': 'geojson',
+            'type': 'fill',
+            'filter': ['all', ['==', '$type', 'Polygon']],
+            'paint': {
+                'fill-color': '#000000',
+                'fill-opacity': 0.25
+            }
+        });
+
+        map.addLayer({
+            'id': 'geojson-point',
+            'source': 'geojson',
+            'type': 'circle',
+            'filter': ['all', ['==', '$type', 'Point']],
+            'paint': {
+                'circle-radius': 5,
+                'circle-color': '#000000'
+            }
+        });
+
+        var source = map.getSource('geojson');
+
+        var testSetData = function(geojson, cb) {
+            var running = true;
+            requestAnimationFrame(function runner() {
+                if (running) {
+                    source.setData(geojson);
+                    requestAnimationFrame(runner);
+                }
+            });
+
+            measureFramerate(DURATION_MILLISECONDS, function(err, fps) {
+                running = false;
+                cb(err, fps);
+            });
+        };
+
+        map.repaint = true;
+        measureFramerate(DURATION_MILLISECONDS, function(err, fps) {
+            if (err) return evented.fire('error', {error: err});
+            evented.fire('log', {message: 'repaint - ' + formatNumber(fps)});
+            map.repaint = false;
+            testSetData(smallFeatureCollection, function(err, fps) {
+                if (err) return evented.fire('error', {error: err});
+                evented.fire('log', {message: 'setData (small) - ' + formatNumber(fps)});
+                testSetData(largeFeatureCollection, function(err, fps) {
+                    if (err) return evented.fire('error', {error: err});
+                    evented.fire('log', {message: 'setData (large) - ' + formatNumber(fps)});
+                });
+            });
+        });
+    });
+
+    return evented;
+};
```

Comment:
I agree that it would be helpful to see the results of multiple benchmarks on the same page at the same time. That is a presentation issue. We could achieve this modifying the benchmark runner to run multiple benchmarks.

Having simple benchmarks that measure exactly one thing in a statistically significant way is more important than having fewer benchmarks. 

I want the ability to run these benchmarks and compare their results in an automated way. Having a single score per benchmark makes those comparisons possible.